integration-tests/opentelemetry-mongodb-client-instrumentation/src/test/java/io/quarkus/it/opentelemetry/BookResourceTest.java

```diff
@@ -44,51 +46,166 @@ void reset() {
     }
 
     @Test
     void reactiveClient() {
         testInsertBooks("/reactive-books");
+        reset();
+        assertThat(get("/reactive-books").as(bookListType)).hasSize(3);
+        assertTraceAvailable("my-reactive-collection");
+        assertParentChild("my-reactive-collection");
+    }
+
+    @Test
+    void reactiveClientMultipleChain() {
+        testInsertBooks("/reactive-books");
+        reset();
+        assertThat(get("/reactive-books/multiple-chain").as(Long.class)).isEqualTo(3L);
+        assertTraceAvailable("my-reactive-collection");
+        assertParentChild("my-reactive-collection");
+    }
+
+    @Test
+    void reactiveClientMultipleCombine() {
+        testInsertBooks("/reactive-books");
+        reset();
+        assertThat(get("/reactive-books/multiple-combine").as(Long.class)).isEqualTo(3L);
         assertTraceAvailable("my-reactive-collection");
+        assertParentChild("my-reactive-collection");
     }
 
-    private void assertTraceAvailable(String dbCollectionName) {
+    @Test
+    void reactiveClientParentChild() {
+        testInsertBooks("/reactive-books");
+        reset();
+        assertThat(get("/reactive-books").as(bookListType)).hasSize(3);
+        assertTraceAvailable("my-reactive-collection");
+        assertParentChild("my-reactive-collection");
+    }
+
+    @Test
+    void reactiveClientError() {
+        given()
+                .get("/reactive-books/invalid")
+                .then()
+                .assertThat()
+                .statusCode(500);
+        assertTraceAvailable("my-reactive-collection", "$invalidop");
+        assertParentChild("my-reactive-collection");
+    }
+
+    @SuppressWarnings("unchecked")
+    private void assertTraceAvailable(String... commandPart) {
+        await().atMost(Duration.ofSeconds(15L)).untilAsserted(() -> {
+            List<Map<String, Object>> spans = getSpans();
+            Collection<ChildSpanData> mongoSpans = new ArrayList<>();
+            for (Map<String, Object> spanData : spans) {
+                if (spanData.get("attributes") instanceof Map attr) {
+                    var cmd = (String) attr.get("mongodb.command");
+                    if (cmd != null) {
+                        assertThat(cmd).contains(commandPart).contains("books");
+                        var parentSpanContext = (Map<String, Object>) spanData.get("parentSpanContext");
+                        mongoSpans.add(new ChildSpanData(
+                                (String) spanData.get("traceId"),
+                                (String) parentSpanContext.get("spanId"),
+                                (String) spanData.get("spanId")));
+                    }
+                }
+            }
+            assertThat(mongoSpans).as("Mongodb statement was not traced.").isNotEmpty();
+        });
+    }
+
+    @SuppressWarnings("unchecked")
+    private void assertParentChild(String... commandPart) {
         await().atMost(Duration.ofSeconds(30L)).untilAsserted(() -> {
```

Comment:
Please don't assert like this because you might be swallowing the real reason. 
Please try:
`await().atMost(Duration.ofSeconds(30L)).until(() -> getSpans().size() == 1);`
and continue after the spans are received.
The number of spans must always be deterministic.